src/test/java/rx/internal/operators/OnSubscribeCombineLatestTest.java

```diff
@@ -884,4 +865,92 @@ public void onNext(Integer t) {
         assertTrue(latch.await(10, TimeUnit.SECONDS));
     }
 
-}
+    @Test
+    public void testCombineMany() {
+        int n = RxRingBuffer.SIZE * 3;
+        
+        Observable<Integer> source = Observable.just(1);
```

Comment:
@akarnokd this may play very bad "joke" with us because operator just needs to combine any result (first, second, etc) correct amount of times to pass the test! 

Can you please create unique observables as a source? Probably even better with `String`s to prevent effects from mathematical operations (equality of sum without fixed order in this case)